### ISSUE TITLE
`gradgradcheck` misses shape mismatch in OpInfo tests for some loss functions with custom double backward implementation.

### ISSUE THREAD

**Issue by soulitzer** [pytorch/pytorch#78355]
This is clear a bug, as shape of `gg0` has to be same as shape of `self`. I do not know why `gradgradcheck` missed that, CC @albanD .

_Originally posted by @nikitaved in https://github.com/pytorch/pytorch/pull/77852#discussion_r877154959_

cc @ezyang @gchanan @zou3519 @albanD @gqchen @pearu @nikitaved @soulitzer @Lezcano @Varal7 @mruberry

**Comment by mruberry** [comment]
@soulitzer @albanD should this be high priority? Seems like some of the gradgrad formulas or tests may be incorrect?

**Comment by soulitzer** [comment]
Actually the `sum` is correct because grad of grad_output should have the same shape as grad_output which should have the same shape as output. The reason the formula still works without the sum is because autograd engine implicitly sums when the shape is broadcastable.

**Comment by nikitaved** [comment]
@soulitzer , I disagree. `grad_output` is the output of the backward method, which is of the same shape as `self`.
Have a look at
https://github.com/pytorch/pytorch/blob/7313a7a987166b1249cf8b62c6c97e9bbd4ecfc6/tools/autograd/derivatives.yaml#L1752-L1765
